includes/class-amp-post-type-support.php

```diff
@@ -0,0 +1,69 @@
+<?php
+/**
+ * AMP Post type support.
+ *
+ * @package AMP
+ * @since 0.6
+ */
+
+/**
+ * Class AMP_Post_Type_Support.
+ */
+class AMP_Post_Type_Support {
+
+	/**
+	 * Add hooks.
+	 */
+	public static function init() {
+		add_action( 'after_setup_theme', array( __CLASS__, 'add_post_type_support' ), 5 );
+		if ( did_action( 'after_setup_theme' ) ) {
+			self::add_post_type_support();
+		}
+	}
+
+	/**
+	 * Get post types that plugin supports out of the box (which cannot be disabled).
+	 *
+	 * @return string[] Post types.
+	 */
+	public static function get_builtin_supported_post_types() {
+		return array_filter( array( 'post' ), 'post_type_exists' );
+	}
+
+	/**
+	 * Get post types that are eligible for AMP support.
+	 *
+	 * @since 0.6
+	 * @return string[] Post types eligible for AMP.
+	 */
+	public static function get_eligible_post_types() {
+		return array_merge(
+			self::get_builtin_supported_post_types(),
+			array_values( get_post_types(
+				array(
+					'public'   => true,
+					'_builtin' => false,
+				),
+				'names'
+			) )
+		);
+	}
+
+	/**
+	 * Declare custom post types support.
+	 *
+	 * This function should only be invoked through the 'after_setup_theme' action to
+	 * allow plugins/theme to overwrite the post types support.
+	 *
+	 * @since 0.6
+	 */
+	public static function add_post_type_support() {
+		$post_types = array_merge(
+			self::get_builtin_supported_post_types(),
+			array_keys( array_filter( AMP_Options_Manager::get_option( 'supported_post_types', array() ) ) )
```

Comment:
True. Eventually I think we should just store an array of post types rather than an array mapping post type names to `true`. In fact, there's no reason why we can't do that now.